js/index.js

```diff
@@ -80,37 +89,103 @@
     const li = document.createElement('li');
     const deleteButton = document.createElement('button');
-    const span = document.createElement('span');
+    const personaNameSpan = document.createElement('span');
+    const input = document.createElement('input');
+    const editPersonButton = document.createElement('button');
+    const savePersonButton = document.createElement('button');
+    const cancelPersonButton = document.createElement('button');
+
+    li.classList.add("person-class");
+    personaNameSpan.classList.add("person-name-class");
+    editPersonButton.classList.add("edit-button");
+    deleteButton.classList.add("delete-button");
+    savePersonButton.classList.add("save-button");   
+    cancelPersonButton.classList.add("cancel-button");   
+
+    personaNameSpan.textContent = person.name;
+    deleteButton.textContent = 'Delete';
+    editPersonButton.textContent = 'Edit';
+    savePersonButton.textContent = 'Save';
+    cancelPersonButton.textContent = 'Cancel';
+
+    li.appendChild(personaNameSpan);
+    li.appendChild(editPersonButton);
+    li.appendChild(deleteButton);
+  
+    personListElement.appendChild(li);
 
-    span.textContent = person.name;
-    deleteButton.textContent = 'delete';
+    editPersonButton.addEventListener('click', () => {
+      input.value = person.name;
 
-    li.appendChild(span);
-    li.appendChild(deleteButton);
+      li.replaceChild(input, personaNameSpan);
+      li.replaceChild(savePersonButton, editPersonButton);
+      li.replaceChild(cancelPersonButton, deleteButton);
+    });
+
+    cancelPersonButton.addEventListener('click', () => {
+      input.value = person.name;
+
+      li.replaceChild(personaNameSpan, input);
+      li.replaceChild(editPersonButton, savePersonButton);
+      li.replaceChild(deleteButton, cancelPersonButton);
+    });
+
+    savePersonButton.addEventListener('click', () => {
+      if (input.value === ""){
+        input.value = person.name;
+      }
+
+      person.name = input.value;
+      personaNameSpan.textContent = person.name;
+
+      li.replaceChild(personaNameSpan, input);
+      li.replaceChild(editPersonButton, savePersonButton);
+      li.replaceChild(deleteButton, cancelPersonButton);
+
+      const editPerson = { name : person.name, id : person.id, }; 
+
+      const personIndex = personList.indexOf(person);
+
+      personList.splice(personIndex, 1, editPerson);
+
+      const editPersonOption = document.createElement('option');
+
+      editPersonOption.id = editPerson.id
+      editPersonOption.value =  editPerson.id;
+      editPersonOption.text = editPerson.name;
+
+      const oldPersonOption = document.getElementById(person.id);
+    
+      selectPerson.replaceChild(editPersonOption, oldPersonOption);
+
+      const taskPersonSpan = document.querySelectorAll('[data-name="' + person.id + '"]');
+
+      for(let i=0 ; i < taskPersonSpan.length; i++) {
+        taskPersonSpan[i].textContent = "：" + editPerson.name;
+      }
```

Comment:
@tacutty 
この辺の処理は結構、悩んだ末に書いた感出てるね笑
こんな感じで簡潔に書けそうだね！ポイントは、名前だけ変更できたら良いから、textContentでピンポイントで変更してあげたら良いところかな！
```suggestion
      // グローバル変数の担当者名を変更
      personList.find(p => p.id = person.id).name = input.value;

      // 担当者一覧の担当者名を変更
      personaNameSpan.textContent = input.value;
      
      // セレクトボックスの担当者名を変更
      document.getElementById(person.id).textContent = input.value;

      // 各タスクの担当者名を変更
      const taskPersonSpan = document.querySelectorAll('[data-name="' + person.id + '"]');
      for(let i=0 ; i < taskPersonSpan.length; i++) {
        taskPersonSpan[i].textContent = "：" + input.value;
      }
      
      // 編集モードを終了させる（inputタグやbuttonを元に戻す
      li.replaceChild(personaNameSpan, input);
      li.replaceChild(editPersonButton, savePersonButton);
      li.replaceChild(deleteButton, cancelPersonButton);
```